Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -1010,6 +1012,12 @@ bool CLuaPedDefs::SetElementBoneRotation(lua_State* const luaVM, CClientPed* ent
     return theEntity ? theEntity->SetBoneRotation(static_cast<eBone>(boneId), yaw, pitch, roll) : false;
 }
 
+bool CLuaPedDefs::SetElementBoneRotationQuaternion(lua_State* const luaVM, CClientPed* entity, std::int32_t boneId, float x, float y, float z, float w)
+{
+    CEntity* theEntity = entity->GetGameEntity();
+    return theEntity ? theEntity->SetBoneRotationQuat(static_cast<eBone>(boneId), x, y, z, w) : false;
```

Comment:
`boneId` have to be checked whether it match **eBone** min and max value or not. I would also replace `int` to `uint`.

Reply:
The GetBoneRotationQuat and SetBoneRotationQuat will return false when bone is not found so there's no really need in doing that

Reply:
It is. But at script's level there have to be a corresponding error message, not just a `false` as result.

Reply:
Is there any predefined variable containing bone ids? or function to check if bone is valid? I can't find it

Reply:
I don't think so. But you can find an example of a simple check in `GetPedBonePosition` function. I think it's sufficient at this point. More accurate checks can be added later during the refactoring.

Reply:
Added